pkg/ebpf/c/tracee.bpf.c

```diff
@@ -656,6 +664,345 @@ int tracepoint__sched__sched_process_fork(struct bpf_raw_tracepoint_args *ctx)
     return 0;
 }
 
+// If we won't create these maps, the userspace needs to be kernel version aware as well,
+// otherwise it'll try to interact with the maps and report an error
```

Comment:
I don't understand this comment. Can you please explain?

Reply:
If I won't create the maps, then the userspace will try to access them and receive an error.
Now, the question is whether this error is "ok" (kernel versions below 5.2 etc). Or perhaps we're above 5.2 but it just failed fetching the map.

Reply:
Where does userspace try to access them?

Reply:
unhidden_modules_map -  https://github.com/OriGlassman/tracee/blob/lkm_finder/pkg/ebpf/tracee.go#L1382 - userspace clears the map after each run.. I'm not familiar with any way to clear a (hash) map from kernelspace, prior 5.13 via this helper: https://github.com/torvalds/linux/commit/69c087ba6225b574afb6e505b72cb75242a3d844

modules_from_procfs -  https://github.com/OriGlassman/tracee/blob/lkm_finder/pkg/ebpf/tracee.go#L1388 - userspace clears the map and fill entries from /proc/modules before each kernelspace run.

Reply:
Just to point out that the maps must be created and can't be under 
"#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 20, 0) || defined(CORE)", unlike the queue used for example:
![image](https://user-images.githubusercontent.com/39296766/228517196-b112868d-6b7b-48a2-a943-81a06ab839c7.png)

Reply:
Ok, now I understand. Since the new logic is only intended for kernel 5.2 and above, one can think that we don't need to create these maps for lower kernel versions, but anyway we should create them to avoid getting errors when trying to access these maps from userspace (which is not aware of the kernel version). Is that correct?
If so, better to update this comment to reflect the fact that the logic (and maps) will only be used in kernels above 5.2